mula/scheduler/context/context.py

```diff
@@ -85,6 +84,53 @@ def __init__(self) -> None:
         self.stop_event: threading.Event = threading.Event()
 
         # Repositories
+        if not self.config.database_dsn.startswith("postgresql"):
+            raise Exception("PostgreSQL is the only supported database backend")
+
         datastore = sqlalchemy.SQLAlchemy(self.config.database_dsn)
         self.task_store: stores.TaskStorer = sqlalchemy.TaskStore(datastore)
         self.pq_store: stores.PriorityQueueStorer = sqlalchemy.PriorityQueueStore(datastore)
+
+        # Metrics collector registry
+        self.metrics_registry: CollectorRegistry = CollectorRegistry()
+
+        Info(
+            name="app_settings",
+            documentation="Scheduler configuration settings",
+            registry=self.metrics_registry,
+        ).info(
+            {
+                "monitor_organisations_interval": str(self.config.monitor_organisations_interval),
+                "pq_maxsize": str(self.config.pq_maxsize),
+                "pq_populate_interval": str(self.config.pq_populate_interval),
+                "pq_populate_grace_period": str(self.config.pq_populate_grace_period),
+                "pq_populate_max_random_objects": str(self.config.pq_populate_max_random_objects),
+            }
+        )
+
+        self.metrics_qsize = Gauge(
+            name="scheduler_qsize",
+            documentation="Size of the scheduler queue",
+            registry=self.metrics_registry,
+            labelnames=["scheduler_id"],
+        )
+
+        self.metrics_katalogus = Info(
+            name="app_katalogus",
+            documentation="Katalogus service information",
+            registry=self.metrics_registry,
+        )
+
+        self.metrics_katalogus_cache_expiry = Gauge(
+            name="katalogus_cache_expiry",
+            documentation="Katalogus cache expiry",
+            registry=self.metrics_registry,
+            labelnames=["cache_type"],
+        )
+
+        self.metrics_katalogus_cache_start = Gauge(
+            name="katalogus_cache_start",
+            documentation="Katalogus cache start",
+            registry=self.metrics_registry,
+            labelnames=["cache_type"],
+        )
```

Comment:
I have a hard time understanding how these two metrics can be used. The cache start/expiry is basically just a random time based on when a request was made after the last expiration if I understand it correctly. How would you create a graph based on this that can help diagnose problems or how would you set alerts on a metric like this?

Reply:
I agree with you that it doesn't give any useful information at the moment, and if the request is: knowing what plugins are known to the scheduler at a specific moment for debugging purposes. We might better be off with an endpoint referencing/reporting it. Additionally, I'm not too fond of using caches for this ( because it introduces a new set of problems), I'd rather have the katalogus have more filter options then there won't be a need to know what is present in the scheduler at a given moment. Although this would create more api chattiness from the scheduler to the katalogus.

For now I'll remove the katalogus metric from the openmetrics implementation, and we can pick up the 'freshness' in a different issue.

Reply:
Made separate issue of this https://github.com/minvws/nl-kat-coordination/issues/856